net/nimble/host/mesh/src/health.c

```diff
@@ -99,6 +99,7 @@ static size_t health_get_current(struct bt_mesh_model *mod,
 			fault_count = 0;
 		} else {
 			sys_put_le16(company_id, company_ptr);
+			net_buf_simple_add(msg, fault_count);
```

Comment:
Could you please extend commit msg by explanation why we need that?
if i undertstand it correctly fault_get_cur() gives application a pointer to buffer where faults are written. If it succeed we need to update our os_mbuf with with len of faults array written by application.

By the way, hope application does not exceed maximum faults array size.
Maybe it would be better to have os_mbuf (net_buf) in the callback instead of *u8_t?
@jhedberg  ^^

Reply:
Ah I see that &fault_count in the callback contains maximum bytes which can be written to the buffer but still, worth to consider passing os_mbuf in the callback.

Reply:
I think I didn't go with a net_buf since net_buf_tailroom() wouldn't actually give you how many faults you can add to the buffer, since there needs to be space for more mesh-internal data after that (in particular the MIC).